enhancements/console/dynamic-plugins.md

```diff
@@ -147,6 +147,44 @@ spec:
     basePath: '/'
 ```
 
+In case the plugin needs to communicate with an in-cluster services, it can
+declare the communication in it's `ConsolePlugin` manifest, defining
+`spec.proxyService` array field entry. In there service `name`,
```

Comment:
+1 this is consistent with the possibility to host plugin assets via explicit URL, instead of via current in-cluster service definition (name, namespace, port).

Reply:
👍 I like the idea. 
It also came to my mind that we could have `spec.services` array field which would contain the required fields like `name`, `namespace`, `port` and `basePath` (which are so far common) together with `type` enum field, which would in this case be `Proxy`.   The `spec.services` array field would also contain other optional fields, like mentioned `caCertificate` or `authorize`.
But that might be more clear down the road when we decide to promote `ConsolePlugin` to `v1beta1` or even to `v1`.